lsp-dart-flutter-daemon.el

```diff
@@ -35,6 +36,89 @@
+                     ((and method (symbolp method)) (symbol-name method))
+                     ((stringp method) method))))
+  (unless params
+    (cl-remf args :params))
+  (let ((json (concat "[" (jsonrpc--json-encode args) "]\r\n")))
+    (jsonrpc--log-event conn args 'client)
+    (process-send-string
+     (jsonrpc--process conn)
+     json)))
+
+(cl-defmethod initialize-instance ((conn lsp-dart-flutter-daemon-connection) _slots)
+  (cl-call-next-method)
+  (let ((proc (jsonrpc--process conn)))
+    (when proc
+      (set-process-filter proc #'lsp-dart-flutter-daemon--process-filter))))
+
+(defun lsp-dart-flutter-daemon--json-read-string (s)
+  (if (fboundp 'json-parse-string)
+      (json-parse-string s
+                         :object-type 'plist
+                         :false-object nil
+                         :null-object nil)
+    (let ((json-object-type 'plist)
+          (json-false :json-false)
+          (json-null nil))
+      (json-read-from-string s))))
+
+;; TODO: Greatly reduce this complexity
+(defun lsp-dart-flutter-daemon--process-filter (proc string)
+  "Invoked when a new STRING has arrived from PROC."
+  (when (buffer-live-p (process-buffer proc))
+    (with-current-buffer (process-buffer proc)
+      (let ((should-trim (not (string-match-p (regexp-quote "][") (string-trim string))))
+            (inhibit-read-only t))
+        (goto-char (process-mark proc))
+        (when (string-prefix-p "[" (string-trim string))
+          (--> string
+               string-trim
+               (replace-regexp-in-string (regexp-quote "},{") "}\n{" it nil 'literal)
+               (split-string it "\n")
+               (-map (lambda (response)
+                       (let ((hash-table (json-parse-string (if should-trim (substring response 1 -1) response)
+                                                            :object-type 'hash-table
+                                                            :false-object :json-false
+                                                            :null-object nil))
+                             (json-message (condition-case-unless-debug oops
+                                               (lsp-dart-flutter-daemon--json-read-string (if should-trim
+                                                                                              (substring response 1 -1)
+                                                                                            response))
+                                             (error
+                                              (jsonrpc--warn "Invalid JSON: %S %s"
+                                                             oops (buffer-string))
+                                              nil)))
+                             (conn (process-get proc 'jsonrpc-connection)))
+                         (when json-message
+                           (message "json: %S\nresponse: %S" json-message response)
```

Comment:
I suppose this was only for debugging purposes? otherwise it should print all the time to minibuffer on any dart project

Reply:
Good catch, yeah just for debugging all fine to remove